blocks/user-licenses/user-licenses.js

```diff
@@ -1,5 +1,400 @@
-import { getEnvironmentConfig, getLocaleConfig } from '../../scripts/zemax-config.js';
-import { createTag } from '../../scripts/scripts.js';
+import { getLocaleConfig } from '../../scripts/zemax-config.js';
+import {
+  createTag, createGenericTable, hideModal, showModal,
+} from '../../scripts/scripts.js';
+import execute from '../../scripts/zemax-api.js';
+
+function showUserActionModal(event) {
+  const newProductUserId = event.target.getAttribute('data-new-productuserid');
+  const licenseId = event.target.getAttribute('data-license-id');
+  const nextActionClass = event.target.getAttribute('data-next-action-class');
+  showModal(event);
+  const userActionButton = document.querySelector(`.${nextActionClass}`);
+  userActionButton.setAttribute('data-new-productuserid', newProductUserId);
+  userActionButton.setAttribute('data-license-id', licenseId);
+}
+
+function createUser(event) {
+  hideModal(event);
+  alert('Implement create user');
+}
+
+function showAddUserTable(event) {
+  const licenseId = event.target.getAttribute('data-license-id');
+  const addUserActionButton = document.querySelector('.add-user-action-button');
+  addUserActionButton.setAttribute('data-license-id', licenseId);
+  showModal(event);
+}
+
+async function changeUserForALicense(event) {
+  const contactId = event.target.getAttribute('contactid');
+  const newProductUserId = event.target.getAttribute('data-new-productuserid');
+  const data = await execute('dynamics_change_enduser_license', `&contact_id=${contactId}&new_productuserid=${newProductUserId}`, 'PATCH');
+  if (data !== undefined && data !== null && data.status === 204) {
```

Comment:
don't check for undefined directly. This can be simplified as
```suggestion
  if (data?.status === 204) {
```
In this version, the code checks data.status only if data exists. If data is null or undefined, the right side of the === will be undefined and the equality will not hold, so the code inside the if statement won't run, which achieves the same result as your original code but is more concise.

Reply:
Completed the changes, if this is preferred syntax then it would great to update our JS lint configuration to catch this to ensure consistency across